### ISSUE TITLE
Native Wayland apps don't trigger transparency.

### ISSUE THREAD

**Issue by HackaN** [rockon999/dynamic-panel-transparency#43]
Cool extension, I had to try it because the "Dynamic Top Bar" extension suddenly disappeared from gnome extension site.

I only have one issue with your extension, which I did not have with the "dynamic top bar":
When I open the Overview and then exit back to the desktop the panel is acting as if there is a maximized window on the screen.

To get the top bar to be transparent I have to maximize and then unmaximize a window.

If it helps I am on Gnome Version 3.22.1

**Comment by rockon999** [comment]
This is a bug in v11 that is fixed in the latest release. The release is pending review and should be available in the coming days. You can update from the GitHub to fix this issue immediately.

**Comment by rockon999** [comment]
Also, if you enable 'Remove excess panel styling' under 'Background' in preferences this issue will be temporarily fixed.

**Comment by rockon999** [comment]
v15 is released on Gnome Extensions.

**Comment by HackaN** [comment]
Thanks for the feedback :-)
I tried the "Remove excess panel styling"-tweak and it did solve the issue (as you said), though it removed the panel-graphics from my theme, haha. However I just updated the extension from Gnome Extensions and it works as it should... Except for GTK3 programs.

If I maximize Firefox, Chrome, Spotify etc and then unmaximize the panel fades away as it should. But if I maximize Evolution, Files etc then the panel stays until I switch to an unfocused program window. When the other program window becomes focused THEN the panel fades away.

So that there is a bug for GTK3-programs is just my assumption, but that is the only relation I can see of when it is happening.

Note: I did, as I said, update the extension but it says "v11" in the About-tab on the extension settings. Is it me who missed something?

**Comment by rockon999** [comment]
I forgot to update the extension version in the 'About' page. Whoops. Aside from that, the behaviour you describe is quite odd. Especially considering Firefox is GTK3 now.

What theme are you using?
Have you added any App Tweaks (in preferences)? 
What other extensions do you have installed? (A screenshot of Tweak Tool would be fine)

Sorry for all the questions! It is hard to narrow down exactly what is causing this without more information.

You could try another theme for now and see if that helps.

**Comment by HackaN** [comment]
Hm, ok so then it cannot be a GTK3 issue.

I just ran through some programs to see which ones this issue occurs with...

Issue:
Evolution, Nautilus, Disks, Transmission, Clocks, Gome-Terminal, Gnome-System-Monitor, LIbre Office, Cheese, Epiphany

No Issue:
Firefox, FileZilla, Chrome, GIMP, Viber

I use Arc-Dark as my GTK+ theme and Axiom as my Shell theme (I actually made that, hehe: https://www.gnome-look.org/p/1154707/)

The extension-list is too large to take a decent screenshot of, lots of preinstalled extensions. But the ones I have active are; "Coverflow alt-tab", "Dash to dock", "Dynamic panel transparency", "Random walls" and "User themes". However I can give a screenshot of my machine:

![apricityos_aspen_macbookpro](https://cloud.githubusercontent.com/assets/26646/19540638/2b19a284-9661-11e6-88c5-c9ad45195380.png)

I tried changing themes (both GTK+ and Shell) but the same thing is happening... so I don't think it has anything to do with the themes. I also tried to switch off the other extensions I have on. Didn't make any difference :-/

**Comment by rockon999** [comment]
It is a Wayland issue. I just switched to Wayland and I can reproduce. It doesn't occur on X, however. Ugh. I'll start looking through Wayland and see what the issue is. I'm guessing Firefox, while using GTK3 now, still runs as an X application under Wayland as they do a lot of custom work.

**Comment by HackaN** [comment]
Ah, that could explain why I get a black frame around Firefox when I start it.

**Comment by rockon999** [comment]
I've identified the issue in Wayland causing this and will be pushing a fix as soon as possible. I just have to finish debugging the fix I'm currently working on.

**Comment by HackaN** [comment]
Sweet!

**Comment by rockon999** [comment]
I think the latest should work for you!

**Comment by HackaN** [comment]
Awesome!

After a (bad) system update my Apricity OS was completely without network. Not just Wi-Fi, even the bloody Ethernet port was dead, haha - So I looked around for other promising Gnome-distros (preferably Arch-based) but gave up after a few tries and went with Ubuntu Gnome 16.04 (which I do not reeaally want to run but for the moment will be what powers my MBP).

Here I still have X. Not sure how to switch over to Wayland (or if that would even be a good idea).

Still, great work and awesome feedback!

**Comment by rockon999** [comment]
Glad to hear you're up and running again! I personally run stock Arch on my rMBP.

**Comment by HackaN** [comment]
I looked at stock Arch, but I found the installation process to be a bit tedious, so I never actually installed it. That's what I liked about Apricity OS, it was just clicking a few buttons and it was installed. Maybe I am too used to the installation process on *ubuntu distros, haha
But now I feel like trying it. I love the Arch Wiki and AUR.

**Comment by HackaN** [comment]
I installed another Arch-based distro, haha:

![skarmbild fran 2016-10-27 02-46-31](https://cloud.githubusercontent.com/assets/26646/19763022/afd48454-9c3c-11e6-9fb1-c90ad069632b.png)

The extension from the Gnome extension site still doesn't work, but I removed it, downloaded the ZIP and installed the extension that way and now it works perfectly with all programs! Great job!

In your instructions it says to do Alt+F2 r Enter - But that doesn't work on Wayland. I had to log out and back in again. But anyway, I am so happy to have a dynamic panel again :-D
